astropy/cosmology/core.py

```diff
@@ -50,13 +57,35 @@
 _CosmoT = TypeVar("_CosmoT", bound="Cosmology")
 _FlatCosmoT = TypeVar("_FlatCosmoT", bound="FlatCosmologyMixin")
 
+# dataclass
+dataclass_decorator = (
+    dataclass(frozen=True, repr=True, eq=False, init=True)
+    if not PYTHON_LT_3_10
+    else lambda x: x
+)
+_dataclass_kwargs = dict(slots=True) if sys.version_info >= (3, 10) else dict()
+
 ##############################################################################
 
 
 class CosmologyError(Exception):
     pass
 
 
+@dataclass(frozen=True, **_dataclass_kwargs)
+class _NameField:
+    default: str | None = None
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return self.default
+        return instance._name
```

Comment:
Does this not fit on one line?
```suggestion
        return self.default if instance is None else instance._name
```

Reply:
I added comments instead explaining the different access methods.